jmdaemon/jmdaemon/message_channel.py

```diff
@@ -284,12 +286,31 @@ def privmsg(self, nick, cmd, message, mc=None):
                           "; cannot find on any message channel.")
             return
 
-    def announce_orders(self, orderlist, nick=None, new_mc=None):
+    @classmethod
+    def serialize_fidelity_bond_message(cls, fidelity_bond_proof):
+        nick_sig = fidelity_bond_proof["nick-signature"]
+        nick_sig = b"\xff"*(72 - len(nick_sig)) + nick_sig
+        cert_sig = hextobin(fidelity_bond_proof["certificate-signature"])
+        cert_sig = b"\xff"*(72 - len(cert_sig)) + cert_sig
+        fidelity_bond_data = struct.pack("<72s72s33sH33s32sII",
+            nick_sig,
+            cert_sig,
+            hextobin(fidelity_bond_proof["certificate-pubkey"]),
+            fidelity_bond_proof["certificate-expiry"],
+            hextobin(fidelity_bond_proof["utxo-pubkey"]),
+            hextobin(fidelity_bond_proof["txid"]),
+            fidelity_bond_proof["vout"],
+            fidelity_bond_proof["locktime"]
+        )
+        return base64.b64encode(fidelity_bond_data).decode("ascii")
```

Comment:
How big does this work out to be? I see 144 for first two sigs, then 66 for two pubkeys, 32 for txid and the others are only a few bytes I guess, plus the 4/3 base64 expansion so I'm guessing about 300-350 bytes over the wire?

As long as we're using IRC we have to think about such things ...